src/modules/fancyzones/editor/FancyZonesEditor/Utils/FancyZonesEditorIO.cs

```diff
@@ -0,0 +1,669 @@
+                {
+                    // Target monitor id
+                    string targetMonitorName = argsParts[(int)CmdArgs.TargetMonitorId];
+
+                    // Monitors count
+                    int count = int.Parse(argsParts[(int)CmdArgs.MonitorsCount]);
+                    if (count != App.Overlay.DesktopsCount)
+                    {
+                        MessageBox.Show(Properties.Resources.Error_Invalid_Arguments, Properties.Resources.Error_Message_Box_Title);
+                        ((App)Application.Current).Shutdown();
+                    }
+
+                    double primaryMonitorDPI = 96f;
+
+                    // parse per monitor data
+                    List<MonitorData> monitorData = new List<MonitorData>();
+                    const int monitorArgsCount = 4;
+                    for (int i = 0; i < count; i++)
+                    {
+                        var data = default(MonitorData);
+                        data.Id = argsParts[(int)CmdArgs.MonitorId + (i * monitorArgsCount)];
+                        data.Dpi = int.Parse(argsParts[(int)CmdArgs.DPI + (i * monitorArgsCount)]);
+                        data.X = int.Parse(argsParts[(int)CmdArgs.MonitorLeft + (i * monitorArgsCount)]);
+                        data.Y = int.Parse(argsParts[(int)CmdArgs.MonitorTop + (i * monitorArgsCount)]);
+                        monitorData.Add(data);
+
+                        if (data.X == 0 && data.Y == 0)
+                        {
+                            primaryMonitorDPI = data.Dpi;
+                        }
+                    }
+
+                    var monitors = App.Overlay.Monitors;
+
+                    // update monitors data
+                    double scaleFactor = 96f / primaryMonitorDPI;
+                    foreach (Monitor monitor in monitors)
+                    {
+                        monitor.Scale(scaleFactor);
+
+                        foreach (MonitorData data in monitorData)
+                        {
+                            if (monitor.Device.ScaledBounds.Left == data.X && monitor.Device.ScaledBounds.Top == data.Y)
+                            {
+                                monitor.Device.Id = data.Id;
+                                monitor.Device.Dpi = data.Dpi;
+                                break;
+                            }
+                        }
+                    }
+
+                    // check if all monitors were mapped
+                    foreach (Monitor monitor in monitors)
+                    {
+                        if (monitor.Device.Id == string.Empty || monitor.Device.Id == null)
+                        {
+                            App.ShowExceptionReportMessageBox(ParsingCmdArgsErrorReport(args[1], count, targetMonitorName, monitorData, monitors));
+                            ((App)Application.Current).Shutdown();
+                            return;
+                        }
+                    }
```

Comment:
```suggestion
```
Let's remove it for now since calling Shutdown while starting up causes a crash in another place.